ayup/portfolio/src/main/webapp/script.js

```diff
@@ -54,3 +54,66 @@ async function getRandomFactUsingAsyncAwait() {
   var a = document.getElementById('js-fact-container');
   a.parentNode.replaceChild(newFactContainer, a);
 }
+
+/** Fetches comments from the server and adds them to the DOM. */
+function loadComments() {
+  fetch('/list-comments').then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comment-list');
+
+    // Deletes all of the current commentss
+    while (commentListElement.firstChild) {
+      commentListElement.removeChild(commentListElement.lastChild);
+    }
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+    })
+  });
+}
+
+/** Creates an element that represents a title, a description, a username and its delete button. */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const titleElement = document.createElement('span');
+  titleElement.innerText = comment.title;
+  titleElement.className = "font-weight-bold";
+
+  const descriptionElement = document.createElement('span');
+  descriptionElement.innerText = comment.description;
+
+  const usernameElement = document.createElement('span');
+  usernameElement.innerText = "By: " + comment.username;
+
+  const deleteButtonElement = createDeleteButton()
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment.id);
+
+    // Loads the comments again
+    loadComments()
```

Comment:
Is there any concern here about loading comments that are already deleted, like a race condition? Do we have to wait for the delete POST request to return before we reload comments?

Reply:
What if I convert the function `deleteComment` to an async promise?

Reply:
I converted `deleteComment` into an asynchronous promise and after that function is done `loadComments` runs. That worked for me right now, but I'm still looking for something else that could work better.